File: app/src/main/java/com/example/petrsumobile/schedule/ScheduleWeekAdapter.java
package com.example.petrsumobile.schedule;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.example.petrsumobile.R;
import java.util.ArrayList;


public class ScheduleWeekAdapter extends RecyclerView.Adapter<ScheduleWeekAdapter.ScheduleWeekViewHolder> {

    private Context context;
    private ArrayList<ScheduleWeek> scheduleWeekArrayList;
    private RecyclerView.RecycledViewPool viewPool = new RecyclerView.RecycledViewPool();

    public static class ScheduleWeekViewHolder extends RecyclerView.ViewHolder {

        private TextView textViewScheduleWeek;
        private RecyclerView rvSubItem;

        public ScheduleWeekViewHolder(View itemView) {
            super(itemView);
            this.textViewScheduleWeek = (TextView) itemView.findViewById(R.id.textViewScheduleWeek);
            this.rvSubItem = itemView.findViewById(R.id.recycle_view_schedule_subitem);
        }
    }

    public ScheduleWeekAdapter(Context context, ArrayList<ScheduleWeek> data) {
        this.context = context;
        this.scheduleWeekArrayList = data;
    }

    @NonNull
    @Override
    public ScheduleWeekViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.schedule_list, parent, false);
        return new ScheduleWeekViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final ScheduleWeekViewHolder holder, final int listPosition) {

        ScheduleWeek scheduleWeek = scheduleWeekArrayList.get(listPosition);

        holder.textViewScheduleWeek.setText(scheduleWeek.getNameWeek());

        LinearLayoutManager layoutManager = new LinearLayoutManager(holder.rvSubItem.getContext());
        layoutManager.setInitialPrefetchItemCount(scheduleWeek.getScheduleArrayList().size());

        ScheduleAdapter subItemAdapter = new ScheduleAdapter(context, scheduleWeek.getScheduleArrayList());

        holder.rvSubItem.setLayoutManager(layoutManager);
        holder.rvSubItem.setAdapter(subItemAdapter);
        holder.rvSubItem.setRecycledViewPool(viewPool);

    }

   @Override
    public int getItemCount() {
        return scheduleWeekArrayList.size();
    }
}

File: app/src/main/java/com/example/petrsumobile/schedule/ScheduleWeekAdapter.kt
package com.example.petrsumobile.schedule

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.RecycledViewPool
import com.example.petrsumobile.R
import com.example.petrsumobile.schedule.ScheduleWeekAdapter.ScheduleWeekViewHolder

class ScheduleWeekAdapter : RecyclerView.Adapter<ScheduleWeekViewHolder>() {
    private val viewPool = RecycledViewPool()
    private val scheduleWeekList = mutableListOf<List<Schedule>>()
    private val weekNames = arrayOf("Понедельник", "Вторник", "Среда", "Четверг", "Пятница", "Суббота", "Воскресенье")

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ScheduleWeekViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.schedule_list, parent, false)
        return ScheduleWeekViewHolder(view)
    }

    override fun onBindViewHolder(holder: ScheduleWeekViewHolder, listPosition: Int) {
        val scheduleList = scheduleWeekList[listPosition]
        val subItemAdapter = ScheduleAdapter(scheduleList)

        holder.textViewScheduleWeek.text = weekNames[listPosition]
        holder.rvSubItem.adapter = subItemAdapter
        holder.rvSubItem.setRecycledViewPool(viewPool)
    }

    override fun getItemCount(): Int = scheduleWeekList.size

    fun updateList(list: List<List<Schedule>>) {
        scheduleWeekList.clear()
        scheduleWeekList.addAll(list)
        notifyDataSetChanged()
    }

    class ScheduleWeekViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val textViewScheduleWeek: TextView = itemView.findViewById<View>(R.id.textViewScheduleWeek) as TextView
        val rvSubItem: RecyclerView = itemView.findViewById(R.id.recycle_view_schedule_subitem)
    }
}